File: backend/src/models/warehouse.model.js
const { Schema, model } = require('mongoose');

const warehouseSchema = new Schema(
  {
    userId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    nameAr: { type: String, required: true, trim: true },
    nameEn: { type: String, required: true, trim: true },
    address: { type: String, required: true },
    city: { type: String, required: true },
    phone: { type: String, required: true },
    logo: { type: String, default: null },
    discountRate: { type: Number, default: 4 },
    commissionRate: { type: Number, default: 1 },
    deliveryStartTime: { type: String, default: null },
    deliveryEndTime: { type: String, default: null },
    inventoryUpdateTime: { type: String, default: null },
    averageRating: { type: Number, default: 0 },
    reviewsCount: { type: Number, default: 0 },
    deliveryType: {
      type: String,
      enum: ['self', 'third_party'],
      default: 'self',
    },
    // Opt-in proof-of-delivery: when true, the warehouse can only advance an
    // order to 'delivered' once the pharmacy has uploaded a photo of the
    // shipment seal/stamp ("الختم") - see warehouseOrder.service.js's
    // advanceOrderStatus gate and order.service.js's attachDeliverySealPhoto.
    // Defaults to false so every existing warehouse is completely unaffected;
    // documents predating this field read as false.
    requireDeliverySealPhoto: { type: Boolean, default: false },
    // Per-warehouse order size limits, in USD - the same currency the cart
    // totals in, so no conversion sits between what the pharmacist sees and
    // what's enforced (order.service.js). Both are opt-in: 0 means "no
    // minimum" and null means "no maximum", which is what every existing
    // warehouse gets by default, leaving them unaffected.
    minOrderAmountUsd: { type: Number, default: 0, min: 0 },
    maxOrderAmountUsd: { type: Number, default: null, min: 0 },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: true }
);

warehouseSchema.index({ userId: 1 });
warehouseSchema.index({ city: 1 });

module.exports = model('Warehouse', warehouseSchema);

File: phoenix/lib/features/warehouse_selection/data/models/warehouse_model.dart
class WarehouseModel {
  const WarehouseModel({
    required this.id,
    required this.nameAr,
    required this.nameEn,
    required this.city,
    required this.phone,
    this.logo,
    this.minOrderAmountUsd = 0,
    this.maxOrderAmountUsd,
  });

  final String id;
  final String nameAr;
  final String nameEn;
  final String city;
  final String phone;
  final String? logo;
  // Order-size limits set by the warehouse itself (backend:
  // warehouse.model.js). 0 = no minimum, null = no maximum.
  final num minOrderAmountUsd;
  final num? maxOrderAmountUsd;

  factory WarehouseModel.fromJson(Map<String, dynamic> json) => WarehouseModel(
    id: json['id'] as String,
    nameAr: json['nameAr'] as String,
    nameEn: json['nameEn'] as String,
    city: json['city'] as String,
    phone: json['phone'] as String,
    logo: json['logo'] as String?,
    minOrderAmountUsd: (json['minOrderAmountUsd'] as num?) ?? 0,
    maxOrderAmountUsd: json['maxOrderAmountUsd'] as num?,
  );
}
